kobo/apps/service_health/views.py

```diff
@@ -37,67 +40,73 @@ def get_response(url_):
     return failure, message, content
 
 
-def service_health(request):
+def check_status(
+    service_name: str, check_function: Callable
+) -> Tuple[Optional[str], float]:
     """
-    Return a HTTP 200 if some very basic runtime tests of the application
-    pass. Otherwise, return HTTP 500
+    Check service via callable function.
+    If an exception is raised, return the class name for public consumption.
+    Log the full exception information. This prevents information leakage
     """
-
-    any_failure = False
-
+    error = None
     t0 = time.time()
     try:
-        settings.MONGO_DB.instances.find_one()
-    except Exception as e:
-        mongo_message = repr(e)
-        any_failure = True
-    else:
-        mongo_message = 'OK'
-    mongo_time = time.time() - t0
+        check_function()
+    except Exception as exception:
+        logging.error(
+            f'Service health {service_name} check failure', exc_info=True
+        )
+        error = repr(type(exception).__name__)
+    cache_time = time.time() - t0
+    return error, cache_time
 
-    t0 = time.time()
-    try:
-        Asset.objects.order_by().first()
-    except Exception as e:
-        postgres_message = repr(e)
-        any_failure = True
-    else:
-        postgres_message = 'OK'
-    postgres_time = time.time() - t0
 
-    t0 = time.time()
-    failure, enketo_message, enketo_content = get_response(settings.ENKETO_INTERNAL_URL)
-    any_failure = True if failure else any_failure
-    enketo_time = time.time() - t0
+def service_health(request):
+    """
+    Return a HTTP 200 if some very basic runtime tests of the application
+    pass. Otherwise, return HTTP 500
+    """
+    all_checks = {
+        'Mongo': lambda: settings.MONGO_DB.instances.find_one(),
+        'Postgres': lambda: Asset.objects.order_by().exists(),
+        'Cache': lambda: cache.set('a', True, 1),
+        'Broker': lambda: celery_app.backend.client.ping(),
```

Comment:
To eliminate the Cypress-based login checker, we need a test for the Redis session store